rclpy/src/rclpy/graph.cpp

```diff
@@ -292,11 +290,15 @@ _get_info_by_topic(
   rcl_ret_t ret = rcl_get_info_by_topic(node, &allocator, topic_name, no_mangle, &info_array);
   if (RCL_RET_OK != ret) {
     if (RCL_RET_UNSUPPORTED == ret) {
-      throw RCLError(
-              "Failed to get information by topic. "
-              "Function not supported by RMW_IMPLEMENTATION");
+      throw NotImplementedError(
+              std::string("Failed to get information by topic for ") +
+              std::string(type) +
+              std::string(": function not supported by RMW_IMPLEMENTATION"));
```

Comment:
@gbalke nit, trick: you only need the first operand to be an `std::string`, for all other a temporary `std::string` will be implicitly created.